### ISSUE TITLE
IRC client should not crash on failed decoding.

### ISSUE THREAD

**Issue by jaraco** [jaraco/irc#34]
In the default configuration, if non-UTF-8 is transmitted to the client, it will crash with an error:

```
#!python

Traceback (most recent call last):
  File "irc_bot.py", line 144, in <module>
    i.run()
  File "irc_bot.py", line 120, in run
    self.client.process_forever()
  File "/usr/local/lib/python3.2/dist-packages/irc/client.py", line 267, in process_forever
    self.process_once(timeout)
  File "/usr/local/lib/python3.2/dist-packages/irc/client.py", line 248, in process_once
    self.process_data(i)
  File "/usr/local/lib/python3.2/dist-packages/irc/client.py", line 213, in process_data
    c.process_data()
  File "/usr/local/lib/python3.2/dist-packages/irc/client.py", line 558, in process_data
    for line in self.buffer:
  File "/usr/local/lib/python3.2/dist-packages/irc/buffer.py", line 84, in <genexpr>
    for line in super(DecodingLineBuffer, self).lines())
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x95 in position 96: invalid start byte

```

---
- Bitbucket: https://bitbucket.org/jaraco/irc/issue/34
- Originally reported by: Anonymous
- Originally created at: 2013-12-16T11:58:09.533

**Comment by jaraco** [comment]
line 80 of buffer.py should be
errors = 'replace'

---

Original comment by: Tim L

**Comment by jaraco** [comment]
This IRC library takes the (somewhat progressive) approach of assuming UTF-8 for all input, but also provides a straightforward mechanism for clients to customize the decoding behavior. See the section Decoding Input in the [Overview](/jaraco/irc/overview). The last example gives a one-line example for the supported mechanism for disabling strict decoding. If you add that to your client startup code, you'll have the behavior you seek.

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Nice to see this is already addressed in the documentation but it's a rather non-sane default for the library to crash with common input. would you consider wrapping the decode in a try and printing a warning that some input was omitted due to the strict utf8 setting?

---

Original comment by: Tim L

**Comment by jaraco** [comment]
Good idea.

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
```
#!python

irc.client.ServerConnection.buffer_class.errors = 'ignore'
```

This fixes it and lets connects much faster.

---

Original comment by: [Yamaii](http://bitbucket.org/Yamaii)

**Comment by jaraco** [comment]
Updated changelog and readme reflecting new LenientDecodingLineBuffer. Fixes #34.

→ <<cset ce95c6e18dcf>>

---

Original comment by: Jason R. Coombs
